link.go

```diff
@@ -638,11 +638,17 @@ func (l *link) IssueCredit(credit uint32) error {
 	return nil
 }
 
+func (l *link) detachOnRejectDisp() bool {
+	if l.detachOnDispositionError && (l.receiver == nil && (l.ReceiverSettleMode == nil || *l.ReceiverSettleMode == ModeFirst)) {
```

Comment:
I know you didn't write this but what is this checking? I guess the receiver settle mode _also_ applies to senders?

Reply:
Or is it just trying to ensure that all the 'receiver' fields are set to their default values?

Reply:
From the spec.

> When set at the sender this indicates the desired value for the settlement mode at the receiver. When set at the receiver this indicates the actual settlement mode in use. The receiver SHOULD respect the sender's desired settlement mode if the sender initiates the attach exchange and the receiver supports the desired mode.

This check is saying, if the receiver settlement mode (RSM) is unspecified or ModeFirst, we automatically detach when receiving a rejection disposition.  I've added a comment that explains this.

Reply:
Ah, that makes a lot of sense. Thanks for that.